### PR TITLE
use media queries to make info message responsive

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

Step towards resolving #3053

### Changes:

* uses media queries in both jsx and sass to handle info button differently in small screen sizes

For small screen sizes, we keep the info message absolutely positioned, but instead of the reference parent element with `position: relative` being the info icon itself, we make a new div, a sibling to the icon, with `position: relative`.

This creates a new row, which is where the info message positions itself.

Since the message is absolutely positioned, it overlaps content below (rather than causing the enclosing modal or other element to have to change form).

### Screenshots

wide window/screen (>= 942px):

image

small window/screen (< 942px):

image

### CODE REVIEW

File: test/unit/components/info-button.test.jsx [hunk 1]
```diff
@@ -26,7 +35,7 @@ describe('InfoButton', () => {
                 message="Here is some info about something!"
             />
         );
-        component.find('div.info-button').simulate('click');
+        component.find('div.info-button').simulate('mouseOver');
```
Comment (picklesrus) [inline test/unit/components/info-button.test.jsx]: why did you change this part of the test?  Looks like the message is still expected to be shown on click, right?

Comment (benjiwheeler) [reply test/unit/components/info-button.test.jsx]: Yes, it should be shown on either `click` or `mouseOver`. This test is focused on the behavior of `mouseOut`, and I thought `mouseOver` would be a more common step before `mouseOut` than `click`.

Comment (picklesrus) [reply test/unit/components/info-button.test.jsx]: I was just looking at the diff wrong (missed the part that it collapsed and thought this was in the test after). Sorry about that.

**Comment by picklesrus** [comment]
The code looks good, but I'm not sure we've landed at the right behavior for that on desktop vs mobile. Happy for it to be merged though and see how it feels when we have a bunch of devices around to test on.

**Comment by benjiwheeler** [comment]
> The code looks good, but I'm not sure we've landed at the right behavior for that on desktop vs mobile. Happy for it to be merged though and see how it feels when we have a bunch of devices around to test on.

Agree on both points!
